sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/mathExpressions.scala

```diff
@@ -1325,3 +1325,94 @@ case class BRound(child: Expression, scale: Expression)
     with Serializable with ImplicitCastInputTypes {
   def this(child: Expression) = this(child, Literal(0))
 }
+
+object WidthBucket {
+
+  def computeBucketNumber(value: Double, min: Double, max: Double, numBucket: Long): jl.Long = {
+    if (numBucket <= 0 || numBucket == Long.MaxValue || jl.Double.isNaN(value) || min == max ||
+        jl.Double.isNaN(min) || jl.Double.isInfinite(min) ||
+        jl.Double.isNaN(max) || jl.Double.isInfinite(max)) {
```

Comment:
I think `value` also cannot be infinite? Or it could be actually?

Reply:
At least, Pg can accept the infinity: https://github.com/postgres/postgres/blob/master/src/test/regress/expected/numeric.out#L843-L848
Currently, this PR follows that behaviour.

Reply:
It seems Oracle can accept it: http://sqlfiddle.com/#!4/db9c1/2/0